src/identity/os_release.rs

```diff
@@ -0,0 +1,114 @@
+//! OS version parsing - utility functions
+
+use failure::{bail, format_err, Fallible, ResultExt};
+use std::io::Read;
+use std::{fs, io};
+use serde_json;
+
+/// OS version flag.
+static OS_VERSION_FLAG: &str = "VERSION";
+
+/// Read original os version info from os alpha version json file.
+pub(crate) fn read_original_os_version<T>(file_path: T) -> Fallible<String>
+where
+    T: AsRef<str>,
+{
+    // open the os release file
+    let fpath = file_path.as_ref();
+    let file = fs::File::open(fpath)
+        .with_context(|e| format_err!("failed to open alpha version file {}: {}", fpath, e))?;
+
+    // parse the content
+    let json: serde_json::Value = serde_json::from_reader(file)
+    .expect("failed to parse alpha version file as JSON");
+    let build: String = json.get("build")
+    .expect("alpha version file does not contain 'build' key")
+    .to_string();
+
+    Ok(build)
+
+}
+
+/// Read current os version info from os release file.
```

Comment:
This is OK, though note the source of truth is the OSTree commit metadata. The `os-release` file is mutated using a treefile option, and e.g. if that option stops working for some reason ([as it has in the past](https://github.com/projectatomic/rpm-ostree/pull/1481)) and we somehow ship it, this code will fail.

The easy way to do this is to just do like Zincati and call `rpm-ostree status --json` and parse that. The harder (but arguably cleaner) way is to e.g. use the [ostree](https://docs.rs/ostree) and [glib](https://docs.rs/glib) crates to do this. (Though from quickly browsing [the glib docs](https://docs.rs/glib/0.8.1/glib/variant/struct.Variant.html), I'm not sure if all the bindings are there to get at the commit metadata tuple from the commit variant object).

Doesn't have to be done here though, feel free to just file an issue for now if you prefer!

Reply:
I see, will be using `rpm-ostree status --json` for now
But the latter option sounds better though, would be great if there's rpm-ostree crate for rust